tests/test_auth.py

```diff
@@ -27,10 +27,13 @@ def test_kernel_channels_authenticated(client, authenticated_user):
 @pytest.mark.parametrize("auth_mode", ("noauth", "token", "user"))
 def test_root_auth(auth_mode, client):
     response = client.get("/")
-    expected = 404
-    if auth_mode == "noauth":
-        expected = 200
+    expected = 200
+    content_type = "text/html; charset=utf-8"
+    if auth_mode in ["token", "user"]:
+        expected = 404
```

Comment:
I think it would be better to return a `401` when no login plugin is installed.

Reply:
Yes, but that depends on `auth_config.login_url` if it exists, will always redirect.

Reply:
I would say that's the responsibility of the user, if they set a `login_url` without installing a login plugin, it's a `404`, otherwise it's a `401`?

Reply:
But the `login_url` is in `fps_auth` config and defaults to `/login`. Is there a way to default to `None` and then the `fps_login` plugin modifies the `auth_config` to set the desired URL?